### PR TITLE
Remove some small code repetition on Flashlight mod classes

### PR DESCRIPTION
**Pull request by o-dasher**
Since every Flashlight class `UpdateFlashlightSize` method was doing something similar to:
```cs
protected override void UpdateFlashlightSize(float size)
{
    this.TransformTo(nameof(FlashlightSize), AdjustedSize, FLASHLIGHT_FADE_DURATION);
}
```
I figured out it would be better to have a method that adjusts the flashlight's adjusted size according to the ruleset requirements rather than a `UpdateFlashlightSize` method.
This also fixes multiple code repetitions on `ModFlashlight` derivated class, because the FlashlightSize had to be normalized on the constructor according to each ruleset. Which now are done on the base class on the `OnLoadComplete` method
```cs
FlashlightSize = adjustCurrentSize();
```
this can be seem on this pull request diff on all the constructors.

### CODE REVIEW

File: osu.Game/Rulesets/Mods/ModFlashlight.cs [hunk 1]
```diff
@@ -123,16 +122,22 @@ protected override void LoadComplete()
...
+                    isBreakTime.BindValueChanged(_ => updateFlashlightSize(), true);
                 }
             }
 
-            protected abstract void UpdateFlashlightSize(float size);
+            private Vector2 adjustCurrentSize() => AdjustSize(GetSize());
+
+            private void updateFlashlightSize() => this.TransformTo(nameof(FlashlightSize), adjustCurrentSize(), 800);
+
+            protected abstract Vector2 AdjustSize(float size);
```
Comment (peppy) [inline osu.Game/Rulesets/Mods/ModFlashlight.cs:140]: Probably a good opportunity to add some simple xmldoc here explaining the input and output.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Rulesets/Mods/ModFlashlight.cs [hunk 2]
```diff
@@ -123,16 +122,22 @@ protected override void LoadComplete()
             {
                 base.LoadComplete();
 
-                Combo.ValueChanged += _ => UpdateFlashlightSize(GetSize());
+                FlashlightSize = adjustCurrentSize();
```
Comment (peppy) [inline osu.Game/Rulesets/Mods/ModFlashlight.cs:125]: This is changing behaviour so should be double-checked.

Comment (o-dasher) [reply osu.Game/Rulesets/Mods/ModFlashlight.cs:137]: The behaviour is still the same, the only thing that changes was that this was being done on the constructors of every `ModFlashlight` implementation.

This defines the `FlashlightSize` so `updateFlashlightSize` works properly:
```cs
this.transformTo(nameof(FlashlightSize)...)
```

File: osu.Game/Rulesets/Mods/ModFlashlight.cs (hunk 1 above)
Comment (o-dasher) [reply osu.Game/Rulesets/Mods/ModFlashlight.cs]: I believe done?

File: osu.Game/Rulesets/Mods/ModFlashlight.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Rulesets/Mods/ModFlashlight.cs:137]: Right, misread the variable names.

**Comment by smoogipoo** [comment]
This will need a conflict resolution merge @o-dasher

File: osu.Game/Rulesets/Mods/ModFlashlight.cs [hunk 3]
```diff
@@ -119,21 +118,33 @@ private void load(ShaderManager shaderManager)
 
             private readonly IBindable<bool> isBreakTime = new BindableBool();
 
+            /// <summary>
+            /// Defines the flashlight area's vector, according to a given flashlight size.
```
[the remark this answers is not in the document]
Comment (o-dasher) [reply osu.Game/Rulesets/Mods/ModFlashlight.cs]: I believe this is fixed, i pretty much copy pasted the docs that were written for the equivalent method on taiko's ruleset before this pr

File: osu.Game/Rulesets/Mods/ModFlashlight.cs [hunk 4]
```diff
@@ -119,21 +118,33 @@ private void load(ShaderManager shaderManager)
 
             private readonly IBindable<bool> isBreakTime = new BindableBool();
 
+            /// <summary>
+            /// Returns the aspect ratio-adjusted size of the flashlight.
+            /// This ensures that the size of the flashlight remains independent of ruleset-specific aspect ratio adjustments.
```
Comment (bdach) [inline osu.Game/Rulesets/Mods/ModFlashlight.cs:123]: This comment is only true for taiko's override of this? It cannot be moved into the xmldoc of the general abstract method. Abstract method xmldocs are supposed to tell *consumers*, i.e. ruleset developers, what they should be overriding the method to.

Comment (o-dasher) [reply osu.Game/Rulesets/Mods/ModFlashlight.cs:123]: You are right. maybe:

Creates the draw flashlight area.
e.g: mania draws on the entire width of the playfield and ajusts the height by a given size, so this should return Vector2(DrawWidth, size).

**Pull request closed without merging**